application/controllers/HistoryController.php

```diff
@@ -49,11 +50,15 @@ public function indexAction()
             ]
         );
         $viewModeSwitcher = $this->createViewModeSwitcher($paginationControl, $limitControl, true);
-        $searchBar = $this->createSearchBar($history, [
+        $preserveParams = [
```

Comment:
The actions have all their specific structure:

* Init (params, auth, tabs)
* get db
* setup query
* setup controls
* handle controls
* apply filters
* export
* register controls
* register content

So initializing `$preserveParams` should be at the top and `createSearchBar` where it has been.